### ISSUE TITLE
The hover state of pills is low contrast and difficult to read

### ISSUE THREAD

**Issue by anoadragon453** [element-hq/element-web#22134]
### Steps to reproduce

1. Hover over a pill
2. Notice that the background becomes light, which makes the white text difficult to read

#### What did you expect?

To not have to :eyes: 

#### What happened instead?

![image](https://user-images.githubusercontent.com/1342360/167681342-6b4188a8-b86f-47a9-8376-94a8bc5df16a.png)

It looks like the related change is https://github.com/matrix-org/matrix-react-sdk/pull/6398.

To be clear, I really like the interactivity that hovering over a pill now has! But I think the chosen highlight color should be tweaked.

### Operating system

Arch Linux

### Application version

Element Nightly version: 2022051001, Olm version: 3.2.8

### How did you install the app?

The AUR

### Will you send logs?

No

**Comment by SimonBrandner** [comment]
@gaelledel, do you think a different colour would be more suitable?

**Comment by daniellekirkwood** [comment]
@janogarcia & @germain-gg even with the new colours this is still the case. Is the contrast not being high enough something we missed in the colour pass or something we missed in the design stages?

**Comment by janogarcia** [comment]
@daniellekirkwood @germain-gg I reported it on Figma. Unsure if it didn't get implemented due to introducing breaking changes to existing themes, though.

https://www.figma.com/file/GmiEFTUaqAulfYzou1Rwrz/Review%3A-Germain's-Typography-and-Color-pass?type=design&node-id=96-79&mode=design&t=XODj8JCEXiKJJSbl-0

![Review- Germain's Typography and Color pass – Figma](https://github.com/vector-im/element-web/assets/60790/e0bec148-96cd-44be-a219-6fd1ab602ca2)

**Comment by janogarcia** [comment]
Forgot to add that, on hover, we could just turn the text to primary color, instead of lowering the contrast for the background.

**Comment by daniellekirkwood** [comment]
oh awesome! :D Thanks both

**Comment by germain-gg** [comment]
@janogarcia i'm happy to implement the change above, but I believe we're still missing the color mapping for the hover state.
Or should we keep the `opacity` trick that we're using, but that now passes the color contrast check?

**Comment by janogarcia** [comment]
@germain-gg Would it be possible to [just do what I described above](https://github.com/vector-im/element-web/issues/22134#issuecomment-1641765582)? 

That is, no change in background color at all, just slightly the text color, if possible with a subtle transition (200-300ms). 

Accessibility wise there's no requirement for how much colors should change in contrast between rest and hover state as long as you're using a pointer which shape changes on hovering the element.

**Comment by daniellekirkwood** [comment]
I think this has been fixed with the most recent changes to the colour of Pills. 

Hey, @robintown can you confirm? I think we can close this out when the newest changes release :)

**Comment by robintown** [comment]
Yes, this is fixed!
